agent/src/agent/pipeline/config/expression_parser/condition.py

```diff
@@ -2,9 +2,47 @@
 Validate condition entered by user and transform it to StreamSets expression language
 """
 import re
+import regex
 
 COMPARISON_FUNCTIONS = ['contains', 'startsWith', 'endsWith', 'matches']
 COMPARISON_LITERALS = ['==', '!=']
+# function name: num of arguments
+FUNCTIONS = {
+    'str:regExCapture': 3
```

Comment:
Can we easily add other functions that return string?

Reply:
well, yeah, just type a new function name here. But depending on a function we might need to change other code a bit, like if it takes a value from a record as a second argument, not the first

Reply:
let's add the ones for which we don't need to change the code
str:replace